backend/routes/user.routes.ts

```diff
@@ -0,0 +1,59 @@
+import express from "express";
+const router = express.Router();
+const passport = require("passport");
+const bcrypt = require("bcrypt");
+const User = require("../models/User");
+
+import { DatabaseUserInterface } from "../utils/Interfaces";
+
+router.post("/signup", async (req, res) => {
+	const { username, password } = req?.body;
+	if (
+		!username ||
+		!password ||
+		typeof username !== "string" ||
```

Comment:
I would expect that you would do some sorts of form validation in the front-end.
You don't have to check here it's a string